### PR TITLE
Dns retention

### PR DESCRIPTION
**Pull request by vara2504**
## Description

Changes for making the DNS log configurable

## For PR author

- [ ] Tests for change.
- [ ] If changing pkg/apis/, run `make gen-files`
- [ ] If changing versions, run `make gen-versions`

## For PR reviewers

A note for code reviewers - all pull requests must have the following:

- [ ] Milestone set according to targeted release.
- [ ] Appropriate labels:
  - `kind/bug` if this is a bugfix.
  - `kind/enhancement` if this is a a new feature.
  - `enterprise` if this PR applies to Calico Enterprise only.

### CODE REVIEW

File: api/v1/logstorage_types.go
Comment (rene-dekker) [inline api/v1/logstorage_types.go:148]: Flow logs are called "Flows"
Dns logs are called "DNSLogs"

Maybe we want to rename it to simply "DNS"?
@caseydavenport

Comment (caseydavenport) [reply api/v1/logstorage_types.go:148]: It's a bit of a bummer that they're not consistently named but I think "DNSLogs" is probably a better name for this than "DNS", which is quite vague and less obvious what it might do.

File: pkg/controller/logstorage/logstorage_controller_test.go
Comment (rene-dekker) [inline pkg/controller/logstorage/logstorage_controller_test.go:321]: I don't think this test adds much value, maybe we can simply remove it. There are several places in this repo where we test that the client can get and update resources.

Comment (vara2504) [reply pkg/controller/logstorage/logstorage_controller_test.go]: Done

**Review by rene-dekker — APPROVED** [review]
lgtm

**Comment by marvin-tigera** [comment]
OK, I will merge the pull request when it's ready, leave the commits as is when I merge it, and delete the branch after I've merged it.

**Comment by marvin-tigera** [comment]
I'm sorry but I failed delete the branch after merging the pull request.

**Pull request merged**
